src/sys/socket/mod.rs

```diff
@@ -213,6 +213,16 @@ pub enum SockProtocol {
     Icmp = libc::IPPROTO_ICMP,
     /// ICMPv6 protocol (ICMP over IPv6)
     IcmpV6 = libc::IPPROTO_ICMPV6,
+    /// SCTP ([sctp(7)](https://man7.org/linux/man-pages/man7/sctp.7.html))
+    #[cfg(not(any(
+        target_os = "haiku",
+        target_os = "redox",
+        target_os = "solaris",
+        target_os = "openbsd",
+        target_os = "illumos",
+        target_os = "dragonfly",
+    )))]
```

Comment:
If possible, we prefer not to use `cfg(not)` unless the # of targets that do not have this symbol is quite few:

```suggestion
    #[cfg(any(
        apple_targets,
        linux_android,
        target_os = "freebsd",
        target_os = "netbsd"
    ))]
```

I hope I do not miss any targets:)

Reply:
Yeah, this makes much more sense as whitelist.
Thank you :)